src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+
+            Sensitivity = ctx.Reader.ReadDouble();
+            Host.CheckDecode(Sensitivity >= 0 && Sensitivity <= 100);
+
+            OutputLength = ctx.Reader.ReadInt32();
+            Host.CheckDecode(OutputLength == 3 || OutputLength == 4 || OutputLength == 7);
+
+            ctx.LoadModel<SrCnnEntireModeler, SignatureLoadModel>(host, out AnomalyModel, "SrCnnEntire");
+            Host.CheckDecode(AnomalyModel != null);
+        }
+
+        private static IDataTransform Create(IHostEnvironment env, Options options, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(options, nameof(options));
+            env.CheckValue(input, nameof(input));
+
+            return new SrCnnEntireTransformer(env, options, input).MakeDataTransform(input);
+        }
+
+        internal static SrCnnEntireTransformer Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(LoaderSignature);
+
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            return new SrCnnEntireTransformer(host, ctx);
+        }
+
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, DataViewSchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            Host.Assert(Threshold >= 0 && Threshold <= 1);
+            Host.Assert(BatchSize == -1 || BatchSize >= 12);
+            Host.Assert(Sensitivity >= 0 && Sensitivity <= 100);
+            Host.Assert(OutputLength == 3 || OutputLength == 4 || OutputLength == 7);
+
+            base.SaveColumns(ctx);
+            ctx.Writer.Write(InputColumnName);
+            ctx.Writer.Write(OutputColumnName);
+            ctx.Writer.Write(Threshold);
+            ctx.Writer.Write((Int32)BatchSize);
+            ctx.Writer.Write((Int32)SrCnnDetectMode);
+            ctx.Writer.Write(Sensitivity);
+            ctx.Writer.Write((Int32)OutputLength);
+
+            ctx.SaveModel(AnomalyModel, "SrCnnEntire");
+        }
+
+        private protected override IRowMapper MakeRowMapper(DataViewSchema schema) => new Mapper(this, schema);
+
+        private sealed class Mapper : OneToOneMapperBase
+        {
+            private readonly SrCnnEntireTransformer _parent;
+            private readonly VBuffer<ReadOnlyMemory<Char>> _slotNames;
+            private readonly int _inputColumnIndex;
+
+            public Mapper(SrCnnEntireTransformer parent, DataViewSchema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                if (!inputSchema.TryGetColumnIndex(parent.InputColumnName, out _inputColumnIndex))
```

Comment:
>inputSchema [](start = 21, length = 11)

The `base` constructor checks this, no need to check again.

Reply:
> > inputSchema [](start = 21, length = 11)
> 
> The `base` constructor checks this, no need to check again.

Refactored Mapper to inherit directly from MapperBase to deal with two input columns.